File: frontend/src/services/geocodingService.ts
import { apiClient } from './api-client';

export interface GeocodingSearchResult {
  refId: string;
  display: string;
  address: string;
  name: string;
  lat: number;
  lng: number;
}

export const getAutocomplete = async (text: string, focus?: string): Promise<GeocodingSearchResult[]> => {
  const response = await apiClient.get<GeocodingSearchResult[]>('/geocoding/autocomplete', {
    params: { text, focus },
  });
  return response.data;
};

export const getSearch = async (text: string, focus?: string): Promise<GeocodingSearchResult[]> => {
  const response = await apiClient.get<GeocodingSearchResult[]>('/geocoding/search', {
    params: { text, focus },
  });
  return response.data;
};

File: backend/src/main/java/com/dut/erp/service/GeocodingService.java
package com.dut.erp.service;

import com.dut.erp.dto.response.GeocodingSearchResult;
import java.util.List;

public interface GeocodingService {
  List<GeocodingSearchResult> autocomplete(String text, String focus);
  List<GeocodingSearchResult> search(String text, String focus);
}
